Storage/Storage/CoreData/IterativeStoreMigrator.swift

```diff
@@ -0,0 +1,220 @@
+
+        // Get the persistent store's metadata.  The metadata is used to
+        // get information about the store's managed object model.
+        guard let sourceMetadata = try metadataForPersistentStore(storeType: storeType, at: sourceStore) else {
+            return false
+        }
+
+        // Check whether the final model is already compatible with the store.
+        // If it is, no migration is necessary.
+        guard targetModel.isConfiguration(withName: nil, compatibleWithStoreMetadata: sourceMetadata) == false else {
+            return true
+        }
+
+        // Find the current model used by the store.
+        guard let sourceModel = try model(for: sourceMetadata) else {
+            return false
+        }
+
+        // Get NSManagedObjectModels for each of the model names given.
+        let objectModels = try models(for: modelNames)
+
+        // Build an inclusive list of models between the source and final models.
+        var modelsToMigrate = [NSManagedObjectModel]()
+        var firstFound = false, lastFound = false, reverse = false
+
+        for model in objectModels {
+            if model.isEqual(sourceModel) || model.isEqual(targetModel) {
+                if firstFound {
+                    lastFound = true
+                    // In case a reverse migration is being performed (descending through the
+                    // ordered array of models), check whether the source model is found
+                    // after the final model.
+                    reverse = model.isEqual(sourceModel)
+                } else {
+                    firstFound = true
+                }
+            }
+
+            if firstFound {
+                modelsToMigrate.append(model)
+            }
+
+            if lastFound {
+                break
+            }
+        }
+
+        // Ensure that the source model is at the start of the list.
+        if reverse {
+            modelsToMigrate = modelsToMigrate.reversed()
+        }
+
+        // Migrate through the list
+        for index in 0...(modelsToMigrate.count - 2) {
+            let modelFrom = modelsToMigrate[index]
+            let modelTo = modelsToMigrate[index + 1]
+
+            // Check whether a custom mapping model exists.
+            guard let migrateWithModel = NSMappingModel(from: nil, forSourceModel: modelFrom, destinationModel: modelTo) ??
+                (try? NSMappingModel.inferredMappingModel(forSourceModel: modelFrom, destinationModel: modelTo)) else {
+                    return false
+            }
+
+            // Migrate the model to the next step
+            DDLogWarn("Attempting migration from \(modelNames[index]) to \(modelNames[index + 1])")
+            
+            guard migrateStore(at: sourceStore, storeType: storeType, fromModel: modelFrom, toModel: modelTo, with: migrateWithModel) == true else {
+                return false
+            }
+        }
+
+        return true
+    }
+}
+
+extension CoreDataIterativeMigrator {
```

Comment:
Style nit: since it appears all of the funcs within this extension are `private`, we could just make the extension itself private and then remove all of the `private` declarations from each function.